packages/hardhat-core/src/internal/core/providers/accounts.ts

```diff
@@ -225,47 +224,37 @@ export class LocalAccountsProvider extends ProviderWrapperWithChainId {
     }
   }
 
-  private async _getNonce(address: Buffer): Promise<BN> {
-    const { bufferToHex } = await import("ethereumjs-util");
+  private async _getNonce(address: Buffer): Promise<bigint> {
+    const { bufferToHex } = await import("@ethereumjs/util");
 
     const response = (await this._wrappedProvider.request({
       method: "eth_getTransactionCount",
       params: [bufferToHex(address), "pending"],
     })) as string;
 
-    return rpcQuantityToBN(response);
+    return rpcQuantityToBigInt(response);
   }
 
   private async _getSignedTransaction(
     transactionRequest: RpcTransactionRequest,
     chainId: number,
     privateKey: Buffer
   ): Promise<Buffer> {
-    const { chains } = await import("@ethereumjs/common/dist/chains");
+    // see HH-1038
+    // const { chains } = await import("@ethereumjs/common/dist/chains");
 
     const { AccessListEIP2930Transaction, Transaction } = await import(
       "@ethereumjs/tx"
     );
 
-    const { default: Common } = await import("@ethereumjs/common");
+    const { Common } = await import("@ethereumjs/common");
 
     const txData = {
       ...transactionRequest,
       gasLimit: transactionRequest.gas,
     };
 
-    // TODO: consider changing instances of "london" below to ["latest hardfork"]
-    const common =
-      chains.names[chainId] !== undefined
-        ? new Common({ chain: chainId, hardfork: "london" })
-        : Common.forCustomChain(
-            "mainnet",
-            {
-              chainId,
-              networkId: chainId,
-            },
-            "london"
-          );
+    const common = Common.custom({ chainId, networkId: chainId });
```

Comment:
This is so weird, but I was present in the discord conversation that motivated it, so ok

Reply:
The tl;dr is that there isn't a lot of difference between creating a `new Common` for a known chain id, and creating a custom common.

In the "Simplify common usages" PR, `new Common` is used only in one or two places in `src`, which means that we are creating commons in the same way almost everywhere. I believe this will make things simpler for us in the future.